src/sql-parser/src/ast/defs/value.rs

```diff
@@ -179,23 +179,31 @@ mod test {
 pub enum DataType {
     /// Array
     Array(Box<DataType>),
-    /// Fixed-length character type e.g. CHAR(10)
-    Char(Option<u64>),
-    /// Decimal type with optional precision and scale e.g. DECIMAL(10,2)
-    Decimal(Option<u64>, Option<u64>),
-    /// Floating point with optional precision e.g. FLOAT(8)
-    Float(Option<u64>),
     /// List
     List(Box<DataType>),
     /// Map
     Map {
         key_type: Box<DataType>,
         value_type: Box<DataType>,
     },
-    /// Types whose names don't accept parameters, e.g. INT
-    Other(Ident),
-    /// Variable-length character type e.g. VARCHAR(10)
-    Varchar(Option<u64>),
+    /// Types who don't embed other types, e.g. INT
+    Other { name: ObjectName, typ_mod: Vec<u64> },
```

Comment:
Could you add to the comment to describe `typ_mod`? The git diff makes it clear, but I'm not sure I'd understand without that context!